analyzer/codechecker_analyzer/analyzers/clangsa/ctu_manager.py

```diff
@@ -168,18 +185,24 @@ def func_map_list_src_to_ast(func_src_list, ctu_on_demand):
 
         # On-demand analysis does not require any preprocessing on the source
         # file paths, contrary to AST-dump based.
-        mapped_path = path if ctu_on_demand else ast_dump_path(path)
-
+        mapped_path = path if config.ctu_on_demand else \
+            ast_dump_path(path, config, triple_arch)
         func_ast_list.append(mangled_name + " " + mapped_path)
     return func_ast_list
 
 
-def get_extdef_mapping_cmd(action, config, source, func_map_cmd):
+def get_extdef_mapping_cmd(
+        action, config, source, func_map_cmd, triple_arch, makefile):
```

Comment:
So `makefile` is supposed to be a boolean parameter, not a path to a Makefile? What is its purpose then? Can you explain a bit more? Can we rename it to something that sounds a bit more boolean-like? (e.g. `has_makefile`)

Reply:
Ok, `has_makefile` or `for_makefile_generation` is really better.